src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.cs

```diff
@@ -135,7 +191,8 @@ private void ProduceTags(TaggerContext<TTag> context, DocumentSnapshotSpan spanT
                         //    editorSnapshot.
 
                         var diagnosticSpan = diagnosticData.GetExistingOrCalculatedTextSpan(sourceText)
-                                                           .ToSnapshotSpan(editorSnapshot);
+                                                           .ToSnapshotSpan(diagnosticSnapshot)
+                                                           .TranslateTo(editorSnapshot, SpanTrackingMode.EdgeExclusive);
```

Comment:
Would it make sense to bump _diagnosticIdToTextSnapshot to hold onto the newer editor snapshot? Consider a case where lots of edits are happening and the diagnostic engine is backed up. _Eventually_ that DiagnosticsUpdated will get triggered and update it no matter what, but is it worth doing it here too?

("no" is a perfectly valid answer, just asking.)

Reply:
> Is it worth clearing the entry on final tagger disposal or something? At that point that buffer ain't useful anymore.

The provider stays around forever, and is shared across all taggers.  We don't currently keep track in any way if there are absolutely no taggers around anymore.

Reply:
> Would it make sense to bump _diagnosticIdToTextSnapshot to hold onto the newer editor snapshot? 

Definitely not.  Because then the spans the DiagnosticService has for that ID would not be correct wrt to that snapshot.

Reply:
![image](https://user-images.githubusercontent.com/201340/36051518-400ed8ec-0d9f-11e8-9af9-8b92e549b97b.png)